test/e2e/storage/csi_volumes.go

```diff
@@ -40,96 +39,81 @@ import (
-	csiExternalAttacherClusterRoleName    string = "system:csi-external-attacher"
-	csiDriverRegistrarClusterRoleName     string = "csi-driver-registrar"
-)
-
-type csiTestDriver interface {
-	createCSIDriver()
-	cleanupCSIDriver()
-	createStorageClassTest(node v1.Node) testsuites.StorageClassTest
+// List of testDrivers to be executed in below loop
+var csiTestDrivers = []func() drivers.TestDriver{
+	drivers.InitHostPathCSIDriver,
+	drivers.InitGcePDCSIDriver,
+	drivers.InitGcePDExternalCSIDriver,
 }
 
-var csiTestDrivers = map[string]func(f *framework.Framework, config framework.VolumeTestConfig) csiTestDriver{
-	"hostPath": initCSIHostpath,
-	"gcePD":    initCSIgcePD,
-	// TODO(#70258): this is temporary until we can figure out how to make e2e tests a library
-	"[Feature: gcePD-external]": initCSIgcePDExternal,
+// List of testSuites to be executed in below loop
+var csiTestSuites = []func() testsuites.TestSuite{
+	testsuites.InitVolumesTestSuite,
+	testsuites.InitVolumeIOTestSuite,
+	testsuites.InitVolumeModeTestSuite,
+	testsuites.InitSubPathTestSuite,
+	testsuites.InitProvisioningTestSuite,
 }
 
+// This executes testSuites for csi volumes.
 var _ = utils.SIGDescribe("CSI Volumes", func() {
-	f := framework.NewDefaultFramework("csi-mock-plugin")
+	f := framework.NewDefaultFramework("volumes")
 
 	var (
-		cs        clientset.Interface
-		crdclient apiextensionsclient.Interface
-		csics     csiclient.Interface
 		ns        *v1.Namespace
-		node      v1.Node
 		config    framework.VolumeTestConfig
+		crdclient apiextensionsclient.Interface
 	)
 
 	BeforeEach(func() {
-		cs = f.ClientSet
-		crdclient = f.APIExtensionsClientSet
-		csics = f.CSIClientSet
 		ns = f.Namespace
-		nodes := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
-		node = nodes.Items[rand.Intn(len(nodes.Items))]
 		config = framework.VolumeTestConfig{
 			Namespace: ns.Name,
 			Prefix:    "csi",
-			// TODO(#70259): this needs to be parameterized so only hostpath sets node name
-			ClientNodeName:    node.Name,
-			ServerNodeName:    node.Name,
-			WaitForCompletion: true,
 		}
-		csiDriverRegistrarClusterRole(config)
-		createCSICRDs(crdclient)
+		drivers.CsiDriverRegistrarClusterRole(config)
+		crdclient = f.APIExtensionsClientSet
+		drivers.CreateCSICRDs(crdclient)
 	})
 
-	for driverName, initCSIDriver := range csiTestDrivers {
-		curDriverName := driverName
-		curInitCSIDriver := initCSIDriver
-
-		Context(fmt.Sprintf("CSI plugin test using CSI driver: %s", curDriverName), func() {
-			var (
-				driver csiTestDriver
-			)
+	for _, initDriver := range csiTestDrivers {
+		curDriver := initDriver()
+		Context(fmt.Sprintf(drivers.GetDriverNameWithFeatureTags(curDriver)), func() {
+			driver := curDriver
 
 			BeforeEach(func() {
-				driver = curInitCSIDriver(f, config)
-				driver.createCSIDriver()
+				// setupDriver
+				drivers.SetCommonDriverParameters(driver, f, config)
+				driver.CreateDriver()
```

Comment:
@msau42 

Thank you for your answer. 
Is my understanding correct that it is also true for all the csi drivers including csi-hostpath?

Then, I will add ```[Serial]``` to test execution logic in csi_volumes.go instead of each ```TestDriver```'s ```FeatureTag```.

Reply:
@pohly is in the process of making hostpath parallelizable, so only gce pd needs to be serialized.